controlplane/kubeadm/internal/machine_collection.go

```diff
@@ -56,14 +56,21 @@ func NewFilterableMachineCollectionFromMachineList(machineList *clusterv1.Machin
 }
 
 // Insert adds items to the set.
-func (s FilterableMachineCollection) Insert(machines ...*clusterv1.Machine) FilterableMachineCollection {
+func (s FilterableMachineCollection) Insert(machines ...*clusterv1.Machine) {
 	for i := range machines {
 		if machines[i] != nil {
 			m := machines[i]
 			s[m.Name] = m
 		}
 	}
-	return s
+}
+
+// Sub returns a copy without machines that are in the given collection
+func (s FilterableMachineCollection) Sub(machines FilterableMachineCollection) FilterableMachineCollection {
```

Comment:
The Kubernetes Set generators use `Difference`, might be good to stick with that convention: https://godoc.org/k8s.io/apimachinery/pkg/util/sets

Reply:
Yeah I actually looked up Python's set library and saw they call it `difference` too but I didn't want to get too esoteric. I'll change it since it's good precedent.